### DISCUSSION TITLE
Ghost counter increments

### DISCUSSION THREAD

**Discussion by Martzipan** [arendst/Tasmota#22337] [Support request]
Hello everyone,

I'm not new but not advanced in ESP. i have the following configuration for an ESP32 DevKit:

Tasmota 14.1.0
GPIOs 16, 17, 18, 19, 21, 22 controlling a 6 relay module connected to 6 24V AC valves for irrigation.
GPIO25 - AM2301
GPIO4 - a switch detached from relays, connected to a rain sensor (the one that closes/open the circuit based on some sponge like material inside)
GPIO26 - a counter connected to a tipping bucket using reed sensor.
Relay module powered by ESP's VCC
ESP powered on the VCC pin since I managed to break the USB plug

Other settings:
CounterDebounce 200
Setoption114 ON
{"Interlock":"ON","Groups":"1,2,3,4,5,6"}

The problem is that whenever i start or stop a valve, the counter gets incremented. After trying all kind of combinations of plug/unplug any of the connected pins, the counter only gets incremented whenever there is power to the valves. For test i have set GPIOs 14 and 15 to Counter2 and Counter3 with nothing connected to them, and they do not get incremented
 
Things i tried:
Pullup resistor for GPIO26 (10K and 4K7)
Switched GPIO26 to GPIO4
Set additional power to the 6 relay module (removed the JD-VCC jumper and added external power to it)

The only time the counter does not increment by valve action is disconnecting the power to the valves (relays still switch on/off).

It looks like somehow the 24V AC circuit that's after the relays influences the 5V circuit that controls the relays and further the ESP pins. The relay module is one from Aliexpress with blue boxlike relays.

Any suggestions of what I'm doing wrong?

**Comment by Martzipan** [comment]
To make things even stranger, i connected the reed sensor to an ESP8266 Tasmota with only that sensor connected and only one counter configured on GPIO5 with and without external pull-up.
Turning on/off valves on the ESP32 increases the counter on the ESP8266.
I changed the GPIO to Switch and Button and the ghosting does not happen.
The reed is ~2m away from valves and ~1m away from relays.
I also tried connecting the 2 boards to different plugs, different wifi APs.

I'm completely lost, pease help!

**Comment by nikito7** [comment]
You may need optocouplers or some filtering/isolation

**Reply by Martzipan** [reply]
I am using an Aliexpress module with optocouplers. What should i filter/isolate? Crossed my mind to wrap the whole rain funnel-bucket thingy in tin foil :D. Although i don't think the reed is influenced in any way since i got ghost increments with it disconnected.

I am thinking of uploading a code to monitor GPIOs but still not sure what should i be looking for. And why is it behaving differently when that GPIO is set to switch or button?

**Reply by nikito7** [reply]
Filter DC power supply.

Maybe a ground issue.

Or you need a stronger pullup.

In esp8266 only gpio 4 and 5 are reliable.

In esp32 I don't know

**Reply by Martzipan** [reply]
i am using 2 different phone chargers to power both via USB.
Not sure how to identify or fix a ground issue. The ESP8266 is a completely different circuit than the ESP32, still gets influenced by it
Tried 4K7 and 10K ohm pull-ups, and again, configuring the same GPIO as switch or button does not get influenced by the relays.

**Comment by Martzipan** [comment]
I flashed a program that reads the value of the reed pin on the A0 every 10ms. i could not make it go below 1000 whatever valves i started/stopped on the other ESP (GPIO5 connected to A0 and reed sensor). I still got increments on actual reed normal action.
Reflashed Tasmota, set the pin to Button and added a relay. I would assume that whenever the pin goes to low, it will toggle the relay (and published over mqtt). Again, whatever actions did to valves had no impact on the button/relay. 
Changed back to Counter ... increments on starting/stopping valves on the other ESP.

So my conclusion is that the pin never goes to LOW unless reed sets it, so there's no magnetic/electric influence when operating the valves or I'm somehow not catching them.

At this point what are the chances of being a Tasmota issue? How could one Tasmota influence another?

**Reply by nikito7** [reply]
Try with counter_n

**Reply by Martzipan** [reply]
tried also with counter_n with and without external pullup. same behaviour.
